### PR TITLE
 Add support for BinaryView in arrow_string::length 

### LINKED ISSUE #6358: Add support for BinaryView in arrow_string::length
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Add support for binary view in the length fn. This was brought up in a code review in #6345

**Describe the solution you'd like**

Code, tests added for calculating length for binary view.

**Describe alternatives you've considered**

Do not implement.

**Additional context**

### PR DESCRIPTION
**Pull request by Omega359**
# Which issue does this PR close?

Closes  #6358 

# Rationale for this change
 

Adding additional functionality 

# What changes are included in this PR?

Code, test.

# Are there any user-facing changes?

No.

### CODE REVIEW

File: arrow-string/src/length.rs [hunk 1]
```diff
@@ -85,6 +85,14 @@ pub fn length(array: &dyn Array) -> Result<ArrayRef, ArrowError> {
         DataType::FixedSizeBinary(len) | DataType::FixedSizeList(_, len) => Ok(Arc::new(
             Int32Array::new(vec![*len; array.len()].into(), array.nulls().cloned()),
         )),
+        DataType::BinaryView => {
+            let list = array.as_binary_view();
+            let v = list.views().iter().map(|v| *v as i32).collect::<Vec<_>>();
```
Comment (alamb) [inline arrow-string/src/length.rs:90]: nice!

File: arrow-string/src/length.rs [hunk 2]
```diff
@@ -224,6 +232,18 @@ mod tests {
         length_binary_helper!(i64, Int64Array, length, value, result)
     }
 
+    #[test]
+    fn length_test_binary_view() {
+        let value: Vec<&[u8]> = vec![b"zero", &[0xff, 0xf8], b"two"];
```
Comment (alamb) [inline arrow-string/src/length.rs:237]: Can you please add a test value that has more than 12 bytes (which ends up running a different code path)?

**Review by alamb — APPROVED** [review]
Thanks @Omega359  -- I have a test suggestion but the code in this PR looks good.

Thank you for making it small and easy to review

File: arrow-string/src/length.rs (hunk 2 above)
Comment (Omega359) [reply arrow-string/src/length.rs:237]: Of course

Comment (Omega359) [reply arrow-string/src/length.rs]: Added

File: arrow-string/src/length.rs [hunk 3]
```diff
@@ -253,6 +261,23 @@ mod tests {
         length_binary_helper!(i64, Int64Array, length, value, result)
     }
 
+    #[test]
+    fn length_test_binary_view() {
+        let value: Vec<&[u8]> = vec![
+            b"zero",
+            &[0xff, 0xf8],
+            b"two",
+            b"this is a longer string to test binary array with",
```
Comment (alamb) [inline arrow-string/src/length.rs:270]: ❤️

**Review by alamb — APPROVED** [review]
Thanks again @Omega359

**Pull request merged**